### PR TITLE
Signals: filter_changed from ProjectListFilter now is emitted when it's on scene tree...

### LINKED ISSUE #36768: Project manager crash on master
**Godot version:**
20595287b

**Issue description:**
When you run Godot, you get this error:
```
[0] ProjectList::set_order_option (x:\inne\godot\source\godot\editor\project_manager.cpp:1405)
[1] ProjectManager::_on_order_option_changed (x:\inne\godot\source\godot\editor\project_manager.cpp:2375)
[2] call_with_variant_args_helper<ProjectManager> (x:\inne\godot\source\godot\core\callable_method_pointer.h:139)
[3] call_with_variant_args<ProjectManager> (x:\inne\godot\source\godot\core\callable_method_pointer.h:161)
[4] CallableCustomMethodPointer<ProjectManager>::call (x:\inne\godot\source\godot\core\callable_method_pointer.h:177)
[5] Callable::call (x:\inne\godot\source\godot\core\callable.cpp:50)
[6] Object::emit_signal (x:\inne\godot\source\godot\core\object.cpp:1200)
[7] Object::emit_signal (x:\inne\godot\source\godot\core\object.cpp:1257)
[8] ProjectListFilter::_filter_option_selected (x:\inne\godot\source\godot\editor\project_manager.cpp:2743)
[9] ProjectListFilter::set_filter_option (x:\inne\godot\source\godot\editor\project_manager.cpp:2737)
[10] ProjectManager::ProjectManager (x:\inne\godot\source\godot\editor\project_manager.cpp:2503)
[11] Main::start (x:\inne\godot\source\godot\main\main.cpp:1931)
[12] widechar_main (x:\inne\godot\source\godot\platform\windows\godot_windows.cpp:160)
[13] _main (x:\inne\godot\source\godot\platform\windows\godot_windows.cpp:184)
[14] main (x:\inne\godot\source\godot\platform\windows\godot_windows.cpp:196)
[15] __scrt_common_main_seh (f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:283)
[16] BaseThreadInitThunk
```
Works if project is edited directly without opening Project Manager.

### PR DESCRIPTION
**Pull request by kuruk-mm**
…to avoid unexpected call

Fixes #36768

### CODE REVIEW

File: editor/project_manager.cpp [hunk 1]
```diff
@@ -2740,7 +2740,8 @@ void ProjectListFilter::_filter_option_selected(int p_idx) {
 	FilterOption selected = (FilterOption)(filter_option->get_selected());
 	if (_current_filter != selected) {
 		_current_filter = selected;
-		emit_signal("filter_changed");
+		if(is_inside_tree())
```
Comment (akien-mga) [inline editor/project_manager.cpp]: Style issue (space between `if` and `(`). See http://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html to set up `clang-format` locally.

Comment (kuruk-mm) [reply editor/project_manager.cpp]: Ups, fixed.

**Review by akien-mga — APPROVED** [review]

**Comment by kuruk-mm** [comment]
I fail a command, and pushed a merge (requesting review from some persons)
I already fixed it, but I cannot remove the reviewers.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**